math/unuran/test/unuranSampler.cxx

```diff
@@ -0,0 +1,234 @@
+// test TUnuran Sampler class using t he DistSampler interface 
+#include "gtest/gtest.h"
+
+#include "Math/DistSampler.h"
+#include "Math/DistSamplerOptions.h"
+#include "Math/Factory.h"
+#include "Math/DistFunc.h"
+#include "Math/Functor.h"
+#include "TH1.h"
+#include "TH2.h"
+
+#include "ROOT/RMakeUnique.hxx"
+
+using namespace ROOT::Math; 
+
+std::unique_ptr<TH1D> FillHisto1D( DistSampler & s) {
+    // create histogram using automatic binning
+    auto h1 = std::make_unique<TH1D>("h1","h1",100, 1, 0);  
+    const int nevt = 10000; 
+    for (int i = 0; i < nevt; i++) {
+        h1->Fill( s.Sample1D());
+    }
+    return h1;
+}
+
+// test using Unuran string API
+TEST(OneDim, StringAPI)
+{
+    std::unique_ptr<DistSampler> sampler(Factory::CreateDistSampler("Unuran"));
+
+    bool ret = sampler->Init("distr = normal(3.,0.75); domain = (0,6) & method = tdr; c = 0");
+    EXPECT_EQ(ret, true);
+    if (!ret) return;
+
+    sampler->SetMode(0.75);
+  
+    auto h1 = FillHisto1D(*sampler);
+    EXPECT_NEAR(h1->GetMean(), 3., 5 * h1->GetMeanError());
+    EXPECT_NEAR(h1->GetRMS(), 0.75, 5*h1->GetRMSError());
+}
+
+// test passing a PDF distribution object
+TEST(OneDim, DistPdfAPI)
+{
+    std::unique_ptr<DistSampler> sampler(Factory::CreateDistSampler("Unuran"));
+
+    auto pdf = [](double x){ return ROOT::Math::normal_pdf(x,0.75,3);};
+    Functor1D pdfDist(pdf);
+    sampler->SetFunction(pdfDist);
+
+    bool ret = sampler->Init("tdr");
+    EXPECT_EQ(ret, true);
+    if (!ret) return;
+
+    sampler->SetRange(0,6.);
+  
+    auto h1 = FillHisto1D(*sampler);
+    EXPECT_NEAR(h1->GetMean(), 3., 5 * h1->GetMeanError());
+    EXPECT_NEAR(h1->GetRMS(), 0.75, 5*h1->GetRMSError());
+}
+
+// test passing a CDF distribution object
+TEST(OneDim, DistCdfAPI)
+{
+    std::unique_ptr<DistSampler> sampler(Factory::CreateDistSampler("Unuran"));
+    auto cdf = [](double x){ return ROOT::Math::normal_cdf(x,0.75,3);};
+    Functor1D cdfDist(cdf);
+    sampler->SetCdf(cdfDist);
+
+    bool ret = sampler->Init("ninv");
+    EXPECT_EQ(ret, true);
+    if (!ret) return;
+
+    sampler->SetRange(0,6.);
+  
+    auto h1 = FillHisto1D(*sampler);
+    EXPECT_NEAR(h1->GetMean(), 3., 5 * h1->GetMeanError());
+    EXPECT_NEAR(h1->GetRMS(), 0.75, 5*h1->GetRMSError());
+}
+
+// test using DistSampler options
+TEST(OneDim, DistSamplerOptAPI){
+
+    std::unique_ptr<DistSampler> sampler(Factory::CreateDistSampler("Unuran"));
+
+    auto pdf = [](double x){ return ROOT::Math::normal_pdf(x,0.75,3);};
+    Functor1D pdfDist(pdf);
+    sampler->SetFunction(pdfDist);
+    DistSamplerOptions opt; 
+    opt.SetAlgorithm("tdr");
+    // set specific options for the tdr method
+    opt.SetAlgoOption("c",0.);
+    opt.SetAlgoOption("cpoints",50);
+    opt.SetAlgoOption("usedars","true");
+    opt.SetAlgoOption("variant_gw","");
+    //opt.Print();
```

Comment:
Does this test check if the algorithm options were correctly propagated to TUnuran? If I understand correctly, this is one of the main developments in this PR.

Reply:
Yes , this is the idea of this test. To check if one can pass the options. Real tests of the algorithms are provided in unuran itself and are outside of the scope of the wrapper classes